pyscriptjs/src/utils.ts

```diff
@@ -109,3 +111,35 @@ export function getAttribute(el: Element, attr: string): string | null {
     }
     return null;
 }
+
+export function calculatePaths(fetch_cfg: FetchConfig[]) {
+    const fetchPaths: string[] = [];
+    const paths: string[] = [];
+    fetch_cfg.forEach(function (each_fetch_cfg: FetchConfig) {
+        const from = each_fetch_cfg.from || "";
+        const to_folder = each_fetch_cfg.to_folder || ".";
+        const files = each_fetch_cfg.files;
+        if (files !== undefined)
+        {
+            for (const each_f of files)
+            {
+                const each_fetch_path = [from.endsWith('/') ? from.slice(0, -1) : from, each_f].filter(it => it!== "").join('/');
+                fetchPaths.push(each_fetch_path);
+                const each_path = [to_folder.endsWith('/') ? to_folder.slice(0, -1) : to_folder, each_f].join('/');
+                paths.push(each_path);
+            }
+        }
+        else
+        {
+            fetchPaths.push(from);
+            const filename = from.split('/').pop();
+            if (filename === '') {
+                throw Error(`Couldn't determine the filename from the path ${from}`);
+            }
+            else {
+                paths.push([to_folder.endsWith('/') ? to_folder.slice(0, -1) : to_folder, filename].join('/'));
```

Comment:
oh, and this is the 3rd time you copy&pasted the code. One more reason to make it a standalone function

Reply:
done